app/src/main/java/com/example/taskyy/ui/screens/LoginScreen.kt

```diff
@@ -58,17 +58,42 @@ fun LoginScreen(state: LoginState, onEvent: (LoginEvent) -> Unit) {
                     .fillMaxHeight(.75f)
                     .padding(30.dp)
             ) {
-                CreateEmailField(isEmailValid = state.isEmailValid, onValueChange =  { email -> onEvent(LoginEvent.OnEmailChanged(email)) }, email =  state.email)
+                state.email?.let {
+                    CreateEmailField(
+                        isEmailValid = state.isEmailValid,
+                        onValueChange = { email -> onEvent(LoginEvent.OnEmailChanged(email)) },
+                        email = it
+                    )
+                }
                 Spacer(modifier = Modifier.height(10.dp))
-                CreatePasswordField(password = state.password, onValueChange = {password -> onEvent(LoginEvent.OnPasswordChanged(password))},
-                    isPasswordVisible =  state.isPasswordVisible,
-                    onClick = {isPasswordVisible -> onEvent(LoginEvent.OnTogglePasswordVisibility(isPasswordVisible))}
-                )
+                state.password?.let {
+                    CreatePasswordField(password = it,
+                        onValueChange = { password -> onEvent(LoginEvent.OnPasswordChanged(password)) },
+                        isPasswordVisible = state.isPasswordVisible,
+                        onClick = { isPasswordVisible ->
+                            onEvent(
+                                LoginEvent.OnTogglePasswordVisibility(
+                                    isPasswordVisible
+                                )
+                            )
+                        }
+                    )
+                }
                 Spacer(modifier = Modifier.height(10.dp))
-                val loginObject = Login(state.email, state.password)
-                TaskyyActionButton(onClick = {onEvent(LoginEvent.OnLoginClick(loginObject))}, text = stringResource(
-                    id = R.string.login
-                ), login = loginObject)
+                val loginObject = state.password?.let { state.email?.let { it1 -> Login(it1, it) } }
+                if (loginObject != null) {
+                    TaskyyActionButton(
+                        onClick = {
+                            loginObject?.let { it1 ->
+                                LoginEvent.OnLoginClick(
+                                    it1
+                                )
+                            }?.let { it2 -> onEvent(it2) }
+                        }, text = stringResource(
+                            id = R.string.login
+                        ), login = loginObject
+                    )
+                }
```

Comment:
Having to construct this login object here seems very unnecessary and overly complex to me given that you have all these `it`, `it1` and `it2` parameters which don't say anything. Why not just pass the email and pw to the VM?

Reply:
That was what I was doing originally but I tried to make my strings nullable in the view model state but then it required all this .let extra stuff to resolve the errors. Then when I tried to get rid of them and use the !! operator instead, the layout of my login screen got messed up.